drivers/sx127x/include/sx127x_registers.h

```diff
@@ -0,0 +1,1710 @@
+#define SX127X_RF_OOKAVG_OOKPEAKTHRESHDEC_010                        0x40
+#define SX127X_RF_OOKAVG_OOKPEAKTHRESHDEC_011                        0x60
+#define SX127X_RF_OOKAVG_OOKPEAKTHRESHDEC_100                        0x80
+#define SX127X_RF_OOKAVG_OOKPEAKTHRESHDEC_101                        0xA0
+#define SX127X_RF_OOKAVG_OOKPEAKTHRESHDEC_110                        0xC0
+#define SX127X_RF_OOKAVG_OOKPEAKTHRESHDEC_111                        0xE0
+
+#define SX127X_RF_OOKAVG_AVERAGEOFFSET_MASK                          0xF3
+#define SX127X_RF_OOKAVG_AVERAGEOFFSET_0_DB                          0x00  /**< Default */
+#define SX127X_RF_OOKAVG_AVERAGEOFFSET_2_DB                          0x04
+#define SX127X_RF_OOKAVG_AVERAGEOFFSET_4_DB                          0x08
+#define SX127X_RF_OOKAVG_AVERAGEOFFSET_6_DB                          0x0C
+
+#define SX127X_RF_OOKAVG_OOKAVERAGETHRESHFILT_MASK                   0xFC
+#define SX127X_RF_OOKAVG_OOKAVERAGETHRESHFILT_00                     0x00
+#define SX127X_RF_OOKAVG_OOKAVERAGETHRESHFILT_01                     0x01
+#define SX127X_RF_OOKAVG_OOKAVERAGETHRESHFILT_10                     0x02  /**< Default */
+#define SX127X_RF_OOKAVG_OOKAVERAGETHRESHFILT_11                     0x03
+
+/**
+ * RegAfcFei
+ */
+#define SX127X_RF_AFCFEI_AGCSTART                                    0x10
+
+#define SX127X_RF_AFCFEI_AFCCLEAR                                    0x02
+
+#define SX127X_RF_AFCFEI_AFCAUTOCLEAR_MASK                           0xFE
+#define SX127X_RF_AFCFEI_AFCAUTOCLEAR_ON                             0x01
+#define SX127X_RF_AFCFEI_AFCAUTOCLEAR_OFF                            0x00  /**< Default */
+
+/**
+ * RegAfcMsb (Read Only)
+ */
+
+/**
+ * RegAfcLsb (Read Only)
+ */
+
+/**
+ * RegFeiMsb (Read Only)
+ */
+
+/**
+ * RegFeiLsb (Read Only)
+ */
+
+/**
+ * RegPreambleDetect
+ */
+#define SX127X_RF_PREAMBLEDETECT_DETECTOR_MASK                       0x7F
+#define SX127X_RF_PREAMBLEDETECT_DETECTOR_ON                         0x80  /**< Default */
+#define SX127X_RF_PREAMBLEDETECT_DETECTOR_OFF                        0x00
+
+#define SX127X_RF_PREAMBLEDETECT_DETECTORSIZE_MASK                   0x9F
+#define SX127X_RF_PREAMBLEDETECT_DETECTORSIZE_1                      0x00
+#define SX127X_RF_PREAMBLEDETECT_DETECTORSIZE_2                      0x20  /**< Default */
+#define SX127X_RF_PREAMBLEDETECT_DETECTORSIZE_3                      0x40
+#define SX127X_RF_PREAMBLEDETECT_DETECTORSIZE_4                      0x60
+
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_MASK                    0xE0
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_0                       0x00
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_1                       0x01
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_2                       0x02
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_3                       0x03
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_4                       0x04
+#define SX127X_RF_PREAMBLEDETECT_DETECTORTOL_5          
```

Comment:
Actually, most part of that registers macro is done by Semtech team in their LoRaMac's sx127x driver sources.